causalpy/pymc_experiments.py

```diff
@@ -336,15 +340,301 @@ def plot(self, counterfactual_label="Counterfactual", round_to=None, **kwargs):
+        cumulative and mean calculations. Default is False.
+
+        Returns:
+        - Dict: A dictionary containing key statistical measures such as posterior estimation,
+        systematic differences, confidence intervals, and posterior MDE for both cumulative and mean results.
+        """
+        results = {}
+        ci = (alpha * 100) / 2
+
+        # Cumulative calculations
+        cumulative_results = self.post_y.sum()
+        _mu_samples_cumulative = (
+            self.post_pred["posterior_predictive"]
+            .mu.stack(sample=("chain", "draw"))
+            .sum("obs_ind")
+        )
+
+        # Mean calculations
+        mean_results = self.post_y.mean()
+        _mu_samples_mean = (
+            self.post_pred["posterior_predictive"]
+            .mu.stack(sample=("chain", "draw"))
+            .mean("obs_ind")
+        )
+
+        # Posterior Mean
+        results["posterior_estimation"] = {
+            "cumulative": round(np.mean(_mu_samples_cumulative.values), 2),
+            "mean": round(np.mean(_mu_samples_mean.values), 2),
+        }
+
+        results["results"] = {
+            "cumulative": round(cumulative_results, 2),
+            "mean": round(mean_results, 2),
+        }
+
+        results["_systematic_differences"] = {
+            "cumulative": results["results"]["cumulative"]
+            - results["posterior_estimation"]["cumulative"],
+            "mean": results["results"]["mean"]
+            - results["posterior_estimation"]["mean"],
+        }
+
+        if correction:
+            _mu_samples_cumulative += results["_systematic_differences"]["cumulative"]
+            _mu_samples_mean += results["_systematic_differences"]["mean"]
+
+        results["ci"] = {
+            "cumulative": [
+                round(np.percentile(_mu_samples_cumulative, ci), 2),
+                round(np.percentile(_mu_samples_cumulative, 100 - ci), 2),
+            ],
+            "mean": [
+                round(np.percentile(_mu_samples_mean, ci), 2),
+                round(np.percentile(_mu_samples_mean, 100 - ci), 2),
+            ],
+        }
+
+        cumulative_upper_mde = round(
+            results["ci"]["cumulative"][1]
+            - results["posterior_estimation"]["cumulative"],
+            2,
+        )
+        cumulative_lower_mde = round(
+            results["posterior_estimation"]["cumulative"]
+            - results["ci"]["cumulative"][0],
+            2,
+        )
+
+        mean_upper_mde = round(
+            results["ci"]["mean"][1] - results["posterior_estimation"]["mean"], 2
+        )
+        mean_lower_mde = round(
+            results["posterior_estimation"]["mean"] - results["ci"]["mean"][0], 2
+        )
+
+        results["posterior_mde"] = {
+            "cumulative": round((cumulative_upper_mde + cumulative_lower_mde) / 2, 2),
+            "mean": round((mean_upper_mde + mean_lower_mde) / 2, 2),
+        }
+        return results
+
+    def power_summary(
```

Comment:
The user-facing methods `power_summary` and `plot_power` are currently methods of the `PrePostFit` class. You can see from the [UML class diagram](https://github.com/pymc-labs/CausalPy/blob/main/CONTRIBUTING.md#overview-of-code-structure) that these methods will be available to both the `SyntheticControl` experiment class _and_ the `Interrupted Time Series` class.

Clearly from the example you are using Bayesian Power Analysis in the context of synthetic control, but can I ask... Does it make sense that these methods are also available in the interrupted time series situations? I don't know the answer, I've not thought about it. But we want to make sure that these methods are placed at the right point in the class hierarchy.

If the Bayesian Power Analysis _is_ also appropriate for the interrupted time series context, then the best possible outcome would be to also create an example notebook for that. But I'd be fine if you instead created a separate issue to deal with that in the near future.

But if this new functionality only makes sense in the context of synthetic control, then can I ask you to either a) move these new methods into the `SyntheticControl` class, or b) create a Bayesian Power Analysis mix-in class instead. I don't mind which, but I think `CausalPy` will probably transition from the current class hierarchy to using mix-ins.